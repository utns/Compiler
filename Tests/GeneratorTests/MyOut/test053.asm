global  main
extern  printf
section .text
foo0:
	push rbp
	mov rbp, rsp
	mov rbx, [v_0]
	mov rax, [rbp + 16]
	mov [rax], rbx
	mov rax, [rbp + 16]
	movq xmm0, [rax]
	mov rax, 1
	mov rdi, formatFloat
	call printf
	xor rax, rax
	mov rdi, formatNewLine
	call printf
	mov rsp, rbp
	pop rbp
	ret
main:
	mov rbx, [v_1]
	mov [v_a], rbx
	push v_a
	call foo0
	add rsp, 8
	movq xmm0, [v_a]
	mov rax, 1
	mov rdi, formatFloat
	sub rsp, 8
	call printf
	add rsp, 8
	xor rax, rax
	mov rdi, formatNewLine
	sub rsp, 8
	call printf
	add rsp, 8
section .data
	formatInt: db "%ld", 0
	formatFloat: db "%f", 0
	formatNewLine: db 10, 0
	v_0: dq 5.200000
	v_a: dq 0.000000
	v_1: dq 4.100000
