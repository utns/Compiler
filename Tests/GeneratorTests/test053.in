procedure foo(var a: float);
begin
    a := 5.2;
    writeln(a);
end;

var
    a: float;
begin
    a := 4.1;
    foo(a);
    writeln(a);
end.
