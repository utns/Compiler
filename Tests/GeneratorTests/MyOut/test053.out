5.200000
5.200000
